### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by gumb0**
Included 2 not yet merged PRs: #443 #546. Didn't include yet: #329.

There was a bunch of CI improvements that I didn't include.

Also not sure if worth mentioning: #521.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,40 @@ Documentation of all notable changes to the **Fizzy** project.
...
+
+- Optimization in `execute` to do single memory allocation for all of arguments, locals and data stack. 
+  [#382](https://github.com/wasmx/fizzy/pull/382)
+- Trap handling in `execute` was simplified. [#425](https://github.com/wasmx/fizzy/pull/425)
+- `fizzy-bench` now uses WABT version 1.0.19. [#443](https://github.com/wasmx/fizzy/pull/443)
+- New simpler implementation of `f32.nearest` and `f64.nearest`. [#504](https://github.com/wasmx/fizzy/pull/504)
+- `execute.hpp` now includes declarations only for `execute` functions. Other functions of C++ API (`instantiate`, 
+  `resolve_imported_functions`, exports access) were moved to `instantiate.hpp`. 
+  [#518](https://github.com/wasmx/fizzy/pull/518)
+- Compiler warning settings now enable more strict checks. [#508](https://github.com/wasmx/fizzy/pull/508)
```
Comment (gumb0) [inline CHANGELOG.md:31]: Not sure how to better describe it or is it worth to mention at all.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,40 @@ Documentation of all notable changes to the **Fizzy** project.
...
+- Parsing module with incorrect memory index now throws `validation_error` instead of `parser_error`.
+  [#546](https://github.com/wasmx/fizzy/pull/546)
+
+### Fixed
+
+- Incorrect results in some floating point operations with NaNs in GCC builds without optimization. 
+  [#513](https://github.com/wasmx/fizzy/pull/513)
+- Potential undefined behaviour in pointer arithmetic in parser.  [#539](https://github.com/wasmx/fizzy/pull/539)
+- Incorrect failure logging in `fizzy-spectests` tool.  [#548](https://github.com/wasmx/fizzy/pull/548)
```
Comment (gumb0) [inline CHANGELOG.md:40]: Not sure if worth to include.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,95 @@ Documentation of all notable changes to the **Fizzy** project.
...
+### Added
+
+- `fizzy-testfloat` tool to check floating-point instructions implementation against Berkeley TestFloat test suite.
+  [#511](https://github.com/wasmx/fizzy/pull/511)
+- `instantiate` function now has `memory_pages_limit` parameter to configure hard limit on memory allocation.
+  `fizzy-spectests` uses the required 4 GB memory limit now.
+  [#515](https://github.com/wasmx/fizzy/pull/515)
+- `fizzy-spectests` tool can now run tests form a single `*.wast` file. [#525](https://github.com/wasmx/fizzy/pull/525)
+- Use the `shellcheck` tool on CI. [#537](https://github.com/wasmx/fizzy/pull/537)
+- There is now `lib/fizzy/cxx` directory with substitute implementations of C++20 standard library features, which Fizzy
```
Comment (chfast) [inline CHANGELOG.md:62]:
```suggestion
- There is now `lib/fizzy/cxx20` directory with substitute implementations of C++20 standard library features, which Fizzy
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,95 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  uses when compiled with pre-C++20 standard (default). It contains now `span` and `bit_cast` utilities.
+  [#535](https://github.com/wasmx/fizzy/pull/535) [#541](https://github.com/wasmx/fizzy/pull/541)
+
+### Changed
+
+- Optimization in `execute` to do allocate memory only once for all of the arguments, locals and data stack.
```
Comment (chfast) [inline CHANGELOG.md:68]:
```suggestion
- Optimization in `execute` to do allocate memory only once for all of the arguments, locals and operand stack.
```

File: CHANGELOG.md (hunk 1 above)
Comment (chfast) [reply CHANGELOG.md:76]: This is interesting item.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,95 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#382](https://github.com/wasmx/fizzy/pull/382)
+- Trap handling in `execute` was simplified. [#425](https://github.com/wasmx/fizzy/pull/425)
+- `fizzy-bench` now uses WABT version 1.0.19. [#443](https://github.com/wasmx/fizzy/pull/443)
+- New simpler implementation of `f32.nearest` and `f64.nearest`. [#504](https://github.com/wasmx/fizzy/pull/504)
+- `execute.hpp` now includes declarations only for `execute` functions. Other functions of C++ API (`instantiate`,
+  `resolve_imported_functions`, exports access) were moved to `instantiate.hpp`.
+  [#518](https://github.com/wasmx/fizzy/pull/518)
+- Compiler warning settings now enable more strict checks. [#508](https://github.com/wasmx/fizzy/pull/508)
+- Parsing module with incorrect memory index now throws `validation_error` instead of `parser_error`.
```
Comment (chfast) [inline CHANGELOG.md:77]:
```suggestion
- Parsing module with incorrect memory/table index now throws `validation_error` instead of `parser_error`.
```

**Review by chfast — COMMENTED** [review]
I would stick to present tense. E.g. "was moved" -> "has been moved".

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,12 +5,114 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#547](https://github.com/wasmx/fizzy/pull/547)
+- Incorrect failure logging in `fizzy-spectests` tool. [#548](https://github.com/wasmx/fizzy/pull/548)
+
 ## [0.4.0] — 2020-09-01
 
 This release introduces complete floating-point support.
 
 With that in place, Fizzy passes all(*) of the [official test suite (spectest 1.0)](https://github.com/wasmx/wasm-spec/tree/w3c-1.0-jsontests-20200813/test/core/json):
-  - 18896 of 18899 binary parser and execution tests,
+  - 17910 of 17913 binary parser and execution tests,
```
Comment (axic) [inline CHANGELOG.md:115]: This number actually included both parser/execution and validation. Separated it given we show validation separately.

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,12 +5,114 @@ Documentation of all notable changes to the **Fizzy** project.
...
+keccak256/512_bytes_rounds_1          105       100        96       162       416        18
+keccak256/512_bytes_rounds_16        1548       100        96       162       418        18
+memset/256_bytes                        7       100        96       155       509        28
+memset/60000_bytes                   1623       100        95       155       507        28
+mul256_opt0/input0                     29       100        91       152       545        22
+mul256_opt0/input1                     29       100        89       152       543        22
+ramanujan_pi/33_runs                  138       100        91       163       483        22
+sha1/512_bytes_rounds_1                94       100        97       181       563        29
+sha1/512_bytes_rounds_16             1318       100        96       179       566        29
+sha256/512_bytes_rounds_1              96       100        98       186       539        30
```
Comment (axic) [inline CHANGELOG.md:61]: I guess we should remove the removed tests?

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,12 +5,114 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  - 989 of 989 validation tests,
+  - 477 skipped due to testing text format parser.
+
+(*) The two failures are caused by, in our opinion, two bugs in the test cases.
+
+A notable change is the upgrade from wabt 1.0.12 to 1.0.19 in our testing infrastructure.
+While this allows us to conduct fuzzing, it must be noted there is a [significant speed regression](https://github.com/WebAssembly/wabt/issues/1550)
+with the newer wabt.
+
+There is a slight speed regression compared to Fizzy 0.4.0, which can be attributed to ... what exactly?
```
Comment (axic) [inline CHANGELOG.md:23]: @chfast what is the reason behind this? Are these number not with the LTO build and the instantiate/execute split is causing it?

File: CHANGELOG.md (hunk 2 above)
Comment (axic) [reply CHANGELOG.md]: I think it is fine.

File: CHANGELOG.md (hunk 3 above)
Comment (axic) [reply CHANGELOG.md:75]: Done.

File: CHANGELOG.md (hunk 4 above)
Comment (axic) [reply CHANGELOG.md:81]: Done.

File: CHANGELOG.md (hunk 5 above)
Comment (axic) [reply CHANGELOG.md:90]: Actually this PR is not merged, will remove the line.

File: CHANGELOG.md (hunk 8 above)
Comment (chfast) [reply CHANGELOG.md:23]: Just micro set is omitted.

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+This release focuses on behind the scenes code and test cleanups, and preparation for a public API.
+
+Fizzy passes all(*) of the [official test suite (spectest 1.0)](https://github.com/wasmx/wasm-spec/tree/w3c-1.0-jsontests-20200813/test/core/json):
+  - 17911 of 17913 binary parser and execution tests,
+  - 989 of 989 validation tests,
+  - 477 skipped due to testing text format parser.
+
+(*) The two failures are caused by, in our opinion, two bugs in the test cases.
+
+A notable change is the upgrade from wabt 1.0.12 to 1.0.19 in our testing infrastructure.
```
Comment (chfast) [inline CHANGELOG.md:19]:
```suggestion
A notable change is the upgrade from WABT 1.0.12 to 1.0.19 in our testing infrastructure.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+While this allows us to conduct fuzzing, it must be noted there is a [significant speed regression](https://github.com/WebAssembly/wabt/issues/1550)
+with the newer wabt.
```
Comment (chfast) [inline CHANGELOG.md:21]:
```suggestion
with the newer WABT.
```

File: CHANGELOG.md (hunk 8 above)
Comment (chfast) [reply CHANGELOG.md:23]: "code layout restructuring in #518".

File: CHANGELOG.md (hunk 7 above)
Comment (chfast) [reply CHANGELOG.md:61]: They may stay in my opinion.

File: CHANGELOG.md [hunk 11]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+
+There is a slight speed regression compared to Fizzy 0.4.0, which can be attributed to ... what exactly?
+
+#### Instantiation
+
+```
+                         fizzy 0.5.0  fizzy 0.4.0   wabt 1.0.19  wasm3 0.4.7
```
Comment (chfast) [inline CHANGELOG.md:28]: I will fix the table layout to match the later one.

File: CHANGELOG.md [hunk 12]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+
+### Added
+
+- `fizzy-testfloat` tool to check floating-point instructions implementation against Berkeley TestFloat test suite.
+  [#511](https://github.com/wasmx/fizzy/pull/511)
+- `instantiate` function now has `memory_pages_limit` parameter to configure hard limit on memory allocation.
+  `fizzy-spectests` uses the required 4 GB memory limit now.
+  [#515](https://github.com/wasmx/fizzy/pull/515)
+- `fizzy-spectests` tool can now run tests form a single `*.wast` file. [#525](https://github.com/wasmx/fizzy/pull/525)
+- Use the `shellcheck` tool on CI. [#537](https://github.com/wasmx/fizzy/pull/537)
```
Comment (chfast) [inline CHANGELOG.md:74]:
```suggestion
- The `shellcheck` linter is being used on CI. [#537](https://github.com/wasmx/fizzy/pull/537)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+- There is now `lib/fizzy/cxx20` directory with substitute implementations of C++20 standard library features, which Fizzy
+  uses when compiled with pre-C++20 standard (default). It contains now `span` and `bit_cast` utilities.
```
Comment (chfast) [inline CHANGELOG.md:76]:
```suggestion
  uses when compiled in pre-C++20 standard (default). It contains now `span` and `bit_cast` utilities.
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#535](https://github.com/wasmx/fizzy/pull/535) [#541](https://github.com/wasmx/fizzy/pull/541)
+
+### Changed
+
+- Optimization in `execute` to do allocate memory only once for all of the arguments, locals and operand stack.
+  [#382](https://github.com/wasmx/fizzy/pull/382)
+- Trap handling in `execute` was simplified. [#425](https://github.com/wasmx/fizzy/pull/425)
```
Comment (chfast) [inline CHANGELOG.md:83]:
```suggestion
- Trap handling in `execute` has been simplified. [#425](https://github.com/wasmx/fizzy/pull/425)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+- `fizzy-bench` now uses WABT version 1.0.19. [#443](https://github.com/wasmx/fizzy/pull/443)
+- New simpler implementation of `f32.nearest` and `f64.nearest`. [#504](https://github.com/wasmx/fizzy/pull/504)
+- `execute.hpp` now includes declarations only for `execute` functions. Other functions of C++ API (`instantiate`,
+  `resolve_imported_functions`, exports access) were moved to `instantiate.hpp`.
```
Comment (chfast) [inline CHANGELOG.md:87]:
```suggestion
  `resolve_imported_functions`, exports access) have been moved to `instantiate.hpp`.
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -5,12 +5,112 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#518](https://github.com/wasmx/fizzy/pull/518)
+- Compiler warning settings now enable more strict checks. [#508](https://github.com/wasmx/fizzy/pull/508)
```
Comment (chfast) [inline CHANGELOG.md:89]:
```suggestion
- More strict compiler warnings have been enabled. [#508](https://github.com/wasmx/fizzy/pull/508)
```

**Review by chfast — APPROVED** [review]

**Pull request merged**